scripts/install_python_prod_dependencies_test.py

```diff
@@ -692,7 +700,9 @@ def test_that_libraries_in_requirements_are_correctly_named(self) -> None:
         # regular letters, digits, periods, underscores, or hyphens and ending
         # with an optional suffix of the pattern [str] with no brackets inside
         # the outside brackets.
-        library_name_pattern = re.compile(r'^[a-zA-Z0-9_.-]+(\[[^\[^\]]+\])*$')
+        library_name_pattern = re.compile(
+            r'^[a-zA-Z0-9_.-]+(\[[^\[^\]]+\])*$|--hash=sha256:[a-zA-Z0-9]+'
```

Comment:
Please make the second part of this regex (`--hash=sha256:[a-zA-Z0-9]+`) match only complete lines by using `^` and `$`, which refer to the start and end of a line, respectively. I think you want something like this for the second part of the regex:

`^\s*--hash=sha256:[a-zA-Z0-9]+$| \\$` (like what you have in `setup.py`)

Reply:
yes done, both regex can be same